R/detect.r

```diff
@@ -49,3 +49,58 @@ str_detect <- function(string, pattern, negate = FALSE) {
     regex = stri_detect_regex(string, pattern, negate = negate, opts_regex = opts(pattern))
   )
 }
+
+#' Detect the presence or absence of a pattern at the beginning or end of a
+#' string.
+#'
+#' Vectorised over `string` and `pattern`. Equivalent to `grepl(paste0("^",
```

Comment:
I think you can just remove the `grepl` equivalent, since it's now more complicated.